src/index.ts

```diff
@@ -6,32 +6,24 @@ import { typeDefs } from './type-defs/type-defs';
 
 const port = 3333;
 
-const server = new ApolloServer({
-  typeDefs,
-  resolvers,
-});
+export async function runDatabase() {
+  await AppDataSource.initialize().catch((error) => {
+    console.info(`[DATABASE] - ${error}`);
+    console.info('[DATABASE] - Shutting down server...');
+    process.exit();
+  });
+}
 
-const startServer = server.listen({ port }).then(({ url }) => {
-  console.log(`[SERVER] - Server running at ${url}`);
-});
+export function runServer() {
+  const server = new ApolloServer({
+    typeDefs,
+    resolvers,
+  });
 
-export async function runServer(isMain = false) {
-  await AppDataSource.initialize()
-    .then(() => {
-      console.log(`[SERVER] - Connected to database.`);
-      startServer;
-    })
-    .catch((error) => {
-      console.log(`[SERVER: ERROR] - ${error}`);
-      server.stop();
-    });
+  runDatabase();
 
-  if (isMain) {
-    startServer;
-  }
+  server.listen({ port });
```

Comment:
Nenhuma dessas duas promises está com await, o que indica que ele vai iniciar essas tarefas emparalelo e vai finalizar a execução dessa função `runServer`. Então se vc for usar essa runServer e tiver que esperar terminar esse setup, vc n tem essa possibilidade. É isso msm que vc queria?